File: src/main/java/com/github/zly2006/reden/ModMenu.java
package com.github.zly2006.reden;

import com.github.zly2006.reden.malilib.GuiConfigs;
import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

public class ModMenu implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return screen -> new GuiConfigs();
    }
}

File: src/main/java/com/github/zly2006/reden/ModMenu.kt
package com.github.zly2006.reden

import com.github.zly2006.reden.malilib.GuiConfigs
import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi

class ModMenu: ModMenuApi {
    override fun getModConfigScreenFactory() = ConfigScreenFactory(::GuiConfigs)
}
